src/OKLogger/Logger.cs

```diff
@@ -56,89 +57,53 @@ public Logger(ILog log, string environment, object context)
             ContextCallbacks = new List<Func<object>>();
         }
 
-
-        /// <summary>
-        /// Generates logging event, adding extra property data
-        /// </summary>
-        /// <param name="level">Fata, Warn, Error, etc.</param>
-        /// <param name="Message">Message to log</param>
-        /// <param name="ex">Exception to log with message</param>
-        private void LogEvent(Level level, IDictionary<string, string> properties, string message, Exception ex)
+        private void LogEvent(Level level, object[] properties, string message, Exception ex)
         {
-
             try
             {
                 if (IsLogLevelEnabled(level))
                 {
-                    var logEvent = new LoggingEvent(typeof(Logger), Log.Logger.Repository, Log.Logger.Name, level, message, ex); ;
-
-                    logEvent.Properties[EnvironmentProperty] = Environment;
-                    //logEvent.Properties[ProcessIdProperty] = Process.GetCurrentProcess().Id;
-
-                    foreach (var contextCallback in ContextCallbacks)
+                    Task.Run(() =>
```

Comment:
IMO, this should be in a separate function called `LogEventAsync` that returns a task, then people could choose to await on it or not.